modules/core/src/lib/layer-state.js

```diff
@@ -7,6 +7,7 @@ export default class LayerState extends ComponentState {
     this.model = null;
     this.needsRedraw = true;
     this.subLayers = null; // reference to sublayers rendered in a previous cycle
+    this.moduleParameters = {};
```

Comment:
The original design of the module parameters system was that one would be able to only supply any new props, and only those uniforms would be generated, avoiding the setting of lots of uniforms. 

Storing and updating a big settings object would seem to totally defeat that.

But perhaps the original `moduleSettings` design doesn't work that well, I recall running into a few issues.